shared/AppInsightsCore/src/JavaScriptSDK/CoreUtils.ts

```diff
@@ -146,6 +183,63 @@ export function objForEachKey(target: any, callbackfn: (name: string, value: any
+ * 
+ * This method is the primary method used to "update" the snippet proxy with the ultimate implementations.
+ * 
+ * Special ES3 Notes:
+ * Updates (setting) of direct property values on the target or indirectly on the source object WILL NOT WORK PROPERLY, updates to the 
+ * properties of "referenced" object will work (target.context.newValue = 10 => will be reflected in the source.context as it's the
+ * same object). ES3 Failures: assigning target.myProp = 3 -> Won't change source.myProp = 3, likewise the reverse would also fail.
+ * @param target - The target object to be assigned with the source properties and functions
+ * @param source - The source object which will be assigned / called by setting / calling the targets proxies
+ * @param chkSet - An optional callback to determine whether a specific property/function should be proxied
+ * @memberof Initialization
+ */
+export function proxyAssign(target: any, source: any, chkSet?: (name: string, isFunc?: boolean, source?: any, target?: any) => boolean) {
+    if (target && source && target !== source && _isObject(target) && _isObject(source)) {
+        // effectively apply/proxy full source to the target instance
+        for (const field in source) {
+            if (CoreUtils.isString(field)) {
+                let value = source[field] as any;
+                if (_isFunction(value)) {
+                    if (!chkSet || chkSet(field, true, source, target)) {
+                        // Create a proxy function rather than just copying the (possible) prototype to the new object as an instance function
+                        target[field as string] = (function(orgFunc: Function) {
+                            return function() {
+                                // Capture the original arguments passed to the method
+                                var originalArguments = arguments;
+                                orgFunc.apply(source, originalArguments);
+                            }
+                        })(value);
+                    }
+                } else if (!chkSet || chkSet(field, false, source, target)) {
+                    if (_hasOwnProperty(target, field)) {
+                        // Remove any previous instance property
+                        delete target[field];
+                    }
+
+                    if (!objDefineAccessors(target, field, () => {
+                        return source[field];
+                    }, (value) => {
+                        source[field] = value;
+                    })) {
+                        // Unable to create an accessor, so just assign the values as a fallback
+                        // -- this will (mostly) work for objects
+                        // -- but will fail for accessing primitives (if the source changes it) and all types of "setters" as the source won't be modified
+                        target[field as string] = value;
```

Comment:
ES3 (IE7/8) fallback -- this is effectively what we did in all previous releases, so there is no issues